src/huntsman/pocs/utils/flats.py

```diff
@@ -0,0 +1,174 @@
+                  "required_exposures": self._required_exposures,
+                  "is_finished": is_finished}
+        return status
+
+    @property
+    def is_finished(self):
+        """ Return True if the exposure sequence is finished, else False.
+        """
+        return self.status["is_finished"]
+
+    def update(self, filename, exptime, time_start):
+        """ Update the sequence data with the previous iteration.
+        Args:
+            average_counts (dict): The average counts.
+            exptime (dict): The exposure times.
+            time_start (datetime.datetime): The time that the exposures were started.
+        """
+        average_counts = self._get_average_counts(filename)
+
+        self._average_counts.append(average_counts)
+        self._times.append(time_start)
+        self._exptimes.append(exptime)
+        self._n_exposures += 1
+
+        if self._validate_exposure(average_counts):
+            self._n_good_exposures += 1
+
+    def get_next_exptime(self, past_midnight):
+        """ Calculate next exptime for flat fields, accounting for changes in sky brightness.
+        Args:
+            past_midnight (bool): True if past midnight (sky is getting brighter), False if not.
+        Returns:
+            astropy.Quantity: The next exposure time.
+        """
+        if self._n_exposures == 0:
+            return self._initial_exposure_time
+
+        elapsed_time = (current_time() - self._times[-1]).sec
+
+        # Get data for specific camera
+        previous_exptime = self._exptimes[-1]
+        previous_counts = self._average_counts[-1]
+
+        # Calculate next exptime
+        exptime = previous_exptime * (self._target_counts / previous_counts)
+        sky_factor = 2.0 ** (elapsed_time / 180.0)
+        if past_midnight:
+            exptime = exptime / sky_factor
+        else:
+            exptime = exptime * sky_factor
+        exptime = exptime.to_value(u.second) * u.second
+
+        # Make sure the exptime is within limits
+        if exptime >= self._max_exptime:
+            self.logger.warning(f"Truncating exptime at maximum value.")
+            exptime = self._max_exptime
+
+        elif exptime <= self._min_exptime:
+            self.logger.warning(f"Truncating exptime at minimum value.")
+            exptime = self._min_exptime
+
+        return exptime
+
+    def _get_average_counts(self, filename, min_counts=1):
+        """ Read the data and calculate a clipped-mean count rate.
+        Args:
+            filename (str): The filename containing the data.
+            min_counts (float): The minimum count rate returned by this function. Can cause
+                problems if less than or equal to 0, so 1 (default) is a safe choice.
+        Returns:
+            float: The average counts.
+        """
+        try:
+            data = self._load_fits_data(filename)
+        except FileNotFoundError:
+            data = self._load_fits_data(filename + ".fz")
```

Comment:
You can be using the `panoptes.utils.images.fits.getdata` to automatically handle this. See below.

```suggestion
        data = self._load_fits_data(filename)
```